test/paginator_test.exs

```diff
@@ -1013,12 +1013,115 @@ defmodule PaginatorTest do
     end
   end
 
+  test "expression based field is passed to cursor_fields" do
+    base_customer_name = "Bob"
+
+    list = create_customers_with_similar_names(base_customer_name)
+
+    {:ok, customer_3} = Enum.fetch(list, 3)
+
+    %Page{entries: entries, metadata: metadata} =
+      base_customer_name
+      |> customers_with_tsvector_rank()
+      |> Repo.paginate(
+        after: encode_cursor(%{rank_value: customer_3.rank_value, id: customer_3.id}),
+        limit: 3,
+        fetch_cursor_value_fun: fn
+          schema, :rank_value ->
+            schema.rank_value
```

Comment:
looks like this case is handled by default fetch_cursor_value_fun implementation

Reply:
This has been removed for both tests.